Bottom: 4951bf3c312c4cecdaef76cf02871ab8890ea72e
Top:    d15e37a8b25977f4c9a9f8142470c0eec73ce1a6
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-23 15:44:46 +0100

Refresh of draw-c-change-drawsilk-to-use-

---

diff --git a/src/draw.c b/src/draw.c
index 0a99bbc..27550fa 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -103,7 +103,7 @@ static void ClearPad (PadTypePtr, bool);
 static void DrawHole (PinTypePtr);
 static void DrawMask (int side, BoxType *);
 static void DrawRats (BoxType *);
-static void DrawSilk (int, int, const BoxType *);
+static void DrawSilk (int side, const BoxType *);
 static int pin_callback (const BoxType * b, void *cl);
 static int pad_callback (const BoxType * b, void *cl);
 
@@ -337,7 +337,7 @@ rat_callback (const BoxType * b, void *cl)
  */
 
 static void
-PrintAssembly (const BoxType * drawn_area, int side_group, int swap_ident)
+PrintAssembly (int side, const BoxType * drawn_area, int side_group)
 {
   int save_swap = SWAP_IDENT;
 
@@ -348,9 +348,7 @@ PrintAssembly (const BoxType * drawn_area, int side_group, int swap_ident)
   gui->set_draw_faded (Output.fgGC, 0);
 
   /* draw package */
-  DrawSilk (swap_ident,
-	    swap_ident ? solder_silk_layer : component_silk_layer,
-	    drawn_area);
+  DrawSilk (side, drawn_area);
   SWAP_IDENT = save_swap;
 }
 
@@ -480,11 +478,12 @@ DrawEverything (BoxTypePtr drawn_area)
   if (gui->set_layer ("soldermask", SL (MASK, BOTTOM), 0))
     DrawMask (SOLDER_LAYER, drawn_area);
 
-  /* Draw top silkscreen */
   if (gui->set_layer ("topsilk", SL (SILK, TOP), 0))
-    DrawSilk (0, component_silk_layer, drawn_area);
+    DrawSilk (COMPONENT_LAYER, drawn_area);
+
   if (gui->set_layer ("bottomsilk", SL (SILK, BOTTOM), 0))
-    DrawSilk (1, solder_silk_layer, drawn_area);
+    DrawSilk (SOLDER_LAYER, drawn_area);
+
   if (gui->gui)
     {
       /* Draw element Marks */
@@ -663,8 +662,10 @@ clearPad_callback (const BoxType * b, void *cl)
  */
 
 static void
-DrawSilk (int new_swap, int layer, const BoxType * drawn_area)
+DrawSilk (int side, const BoxType * drawn_area)
 {
+  int layer = (side == SOLDER_LAYER) ? solder_silk_layer
+                                     : component_silk_layer;
 #if 0
   /* This code is used when you want to mask silk to avoid exposed
      pins and pads.  We decided it was a bad idea to do this
@@ -679,10 +680,8 @@ DrawSilk (int new_swap, int layer, const BoxType * drawn_area)
 #endif
       DrawLayer (LAYER_PTR (layer), drawn_area);
       /* draw package */
-      r_search (PCB->Data->element_tree, drawn_area, NULL, frontE_callback,
-		NULL);
-      r_search (PCB->Data->name_tree[NAME_INDEX (PCB)], drawn_area, NULL,
-		frontN_callback, NULL);
+      r_search (PCB->Data->element_tree, drawn_area, NULL, element_callback, &side);
+      r_search (PCB->Data->name_tree[NAME_INDEX (PCB)], drawn_area, NULL, name_callback, &side);
 #if 0
     }
 
@@ -696,10 +695,8 @@ DrawSilk (int new_swap, int layer, const BoxType * drawn_area)
       gui->use_mask (HID_MASK_AFTER);
       DrawLayer (LAYER_PTR (max_copper_layer + layer), drawn_area);
       /* draw package */
-      r_search (PCB->Data->element_tree, drawn_area, NULL, frontE_callback,
-		NULL);
-      r_search (PCB->Data->name_tree[NAME_INDEX (PCB)], drawn_area, NULL,
-		frontN_callback, NULL);
+      r_search (PCB->Data->element_tree, drawn_area, NULL, element_callback, &side);
+      r_search (PCB->Data->name_tree[NAME_INDEX (PCB)], drawn_area, NULL, name_callback, &side);
     }
   gui->use_mask (HID_MASK_OFF);
 #endif
